azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -291,6 +300,13 @@ def run(self, goal_state):
             if warn_msg != "":
                 self.__log_event(LogLevel.WARNING, warn_msg)
 
+            daemon_version = self.__get_daemon_version_for_update()
+            if requested_version < daemon_version:
+                # Don't process the update if the requested version is less than daemon version,
```

Comment:
also, let's add a test case for this to agent_update

Reply:
sorry, I should have been explicit: I meant add this scenario to your end-to-end tests (can be a separate PR). Having a unit test is also great, thanks for adding it.

Reply:
never mind, it's there